src/main/java/com/mattermost/integration/figma/security/service/OAuthServiceImpl.java

```diff
@@ -74,7 +76,7 @@ public FigmaTokenDTO getFigmaUserToken(InputPayload payload) {
         ResponseEntity<String> resp = restTemplate.postForEntity(url, null, String.class);
         FigmaTokenDTO token = (FigmaTokenDTO) jsonUtils.convertStringToObject(resp.getBody(), FigmaTokenDTO.class).get();
         log.info("Successfully received token");
-        FigmaTokenProvider.token = token;
+        kvService.put(clientId, token, payload.getContext().getMattermostSiteUrl(), payload.getContext().getBotAccessToken());
```

Comment:
please consider to move put logic into subscribe method. Also i'm not sure that we need payload.getContext().getBotAccessToken() in kv as a result that this value is expiring so we can't reuse it when we will use kv.get()

Reply:
payload.getContext().getMattermostSiteUrl() also this stuff we also could have from mm payload